File: java/src/main/java/kata/Roulette.java
package kata;


/**
 * Everything that is hard to test is extracted into collaborators, that can
 * be tested with Focused Integration Tests
 */
public class Roulette {

  private final BallSpinAdapter ball;
  private final Between0And37Randomizer randomizer;

  public Roulette(BallSpinAdapter ball, Between0And37Randomizer randomizer) {
    this.ball = ball;
    this.randomizer = randomizer;
  }

  public Result playGame() {
    ball.spin();
    int randomResult = randomizer.getRouletteResult();
    return new Result(randomResult);
  }

}

File: python/src/kata/roulette.py
from .ball_spin_adapter import BallSpinAdapter
from .between_0_and_37_randomizer import Between0And37Randomizer
from .result import Result

class Roulette:
    def __init__(self, ball: BallSpinAdapter, randomizer: Between0And37Randomizer):
        self.ball = ball
        self.randomizer = randomizer

    def play_game(self) -> Result:
        self.ball.spin()
        random_result = self.randomizer.get_roulette_result()
        return Result(random_result) 